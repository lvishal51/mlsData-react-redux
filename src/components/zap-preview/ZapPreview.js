// @flow

import React, { Component } from 'react';
import './ZapPreview.scss';
import _ from 'lodash';

import {
  Button,
  Table,
  Modal,
  ModalHeader,
  ModalFooter,
  ModalBody
} from 'reactstrap'

type Props = {
  previewTableHeader: Object,
  previewTableData: Object,
  modalClose: Function,
  expression: string
};

type State = {};

export default class ZapPreview extends Component<Props, State> {

  renderContentList = (previewTableHeader: Object, previewTableData: Object) => {
    let contentList = [];
    _.map(previewTableData, (item, index) => {
      contentList.push (
        <tr key={index}>
          <th key={index} scope="row">{index+1}</th>
          {
            _.map(previewTableHeader, (column, key) => {
              return <td key={key}>{item[column]}</td>
            })
          }
        </tr>
      )
    });
    if(_.isEmpty(contentList)) {
      return "No data available for the current mapping"
    } else {
      return contentList;
    }
  }

  render() {
   const { previewTableHeader, previewTableData, modalClose, expression } = this.props;

    return (
      <Modal isOpen={true} toggle={modalClose} className="customModal preview">
      <ModalHeader toggle={modalClose}>Transformation Preview : {expression}</ModalHeader>
      <ModalBody>
        <Table >
          <thead>
            <tr>
              <th>Index</th>
              {_.map(previewTableHeader, (value, key) => { return <th key={key}>{value}</th> })}
            </tr>
          </thead>
          <tbody>
            {
              this.renderContentList(previewTableHeader, previewTableData)
             }
          </tbody>
        </Table>
      </ModalBody>
      <ModalFooter>
        <Button color="primary" onClick={modalClose}>Close</Button>
      </ModalFooter>
    </Modal>
    )
  }
}
