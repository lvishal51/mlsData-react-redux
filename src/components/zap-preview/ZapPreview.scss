.preview.modal-dialog {
  max-width: 1024px;
  top: 5%;
}