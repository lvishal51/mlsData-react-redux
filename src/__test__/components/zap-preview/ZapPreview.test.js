import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import  ZapPreview  from '../../../components/zap-preview/ZapPreview';

  const previewTableData = [ 
     {Sewer: "Public Sewer", ListAgentMlsId: "3217406", ListAgentLastName: "Palatano", Address: "Public Sewer3217406Palatano"},
     {Sewer: "Public Sewer", ListAgentMlsId: "3143777", ListAgentLastName: "Moore", Address: "Public Sewer3143777Moore"},
     {Sewer: "Public Sewer", ListAgentMlsId: "3228726", ListAgentLastName: "Wagner", Address: "Public Sewer3228726Wagner"},
     {Sewer: "Public Sewer", ListAgentMlsId: "3158743", ListAgentLastName: "Masih", Address: "Public Sewer3158743Masih"},
     {Sewer: "Public Sewer", ListAgentMlsId: "78774", ListAgentLastName: "Gatewood", Address: "Public Sewer78774Gatewood"}
  ] 
  const previewTableHeader = ['Sewer','ListAgentMlsId','ListAgentLastName']

const inputData = {
  previewTableData: previewTableData,
  previewTableHeader: previewTableHeader,
  modalClose: () => {}
};

describe('Zap Table component', () => {

  it('should render correctly.', () => {
    const wrapper = shallow(<ZapPreview {...inputData}/>);
    wrapper.instance().forceUpdate();
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('renderContentList function with data should be triggered correctly', () => {
    const wrapper = shallow(<ZapPreview  {...inputData}/>);
    const instance = wrapper.instance();
    instance.renderContentList(previewTableHeader, previewTableData);
  });

  it('renderContentList function with data should be triggered correctly', () => {
    const wrapper = shallow(<ZapPreview  {...inputData}/>);
    const instance = wrapper.instance();
    instance.renderContentList(previewTableHeader, []);
  });

});
