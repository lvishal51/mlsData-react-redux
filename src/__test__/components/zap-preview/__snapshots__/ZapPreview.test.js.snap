// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Zap Table component should render correctly. 1`] = `
<Modal
  autoFocus={true}
  backdrop={true}
  backdropTransition={
    Object {
      "mountOnEnter": true,
      "timeout": 150,
    }
  }
  centered={false}
  className="customModal preview"
  fade={true}
  isOpen={true}
  keyboard={true}
  modalTransition={
    Object {
      "timeout": 300,
    }
  }
  onClosed={[Function]}
  onOpened={[Function]}
  role="dialog"
  toggle={[Function]}
  zIndex={1050}
>
  <ModalHeader
    charCode={215}
    closeAriaLabel="Close"
    tag="h5"
    toggle={[Function]}
    wrapTag="div"
  >
    Transformation Preview : 
  </ModalHeader>
  <ModalBody
    tag="div"
  >
    <Table
      responsiveTag="div"
      tag="table"
    >
      <thead>
        <tr>
          <th>
            Index
          </th>
          <th
            key="0"
          >
            Sewer
          </th>
          <th
            key="1"
          >
            ListAgentMlsId
          </th>
          <th
            key="2"
          >
            ListAgentLastName
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          key="0"
        >
          <th
            key="0"
            scope="row"
          >
            1
          </th>
          <td
            key="0"
          >
            Public Sewer
          </td>
          <td
            key="1"
          >
            3217406
          </td>
          <td
            key="2"
          >
            Palatano
          </td>
        </tr>
        <tr
          key="1"
        >
          <th
            key="1"
            scope="row"
          >
            2
          </th>
          <td
            key="0"
          >
            Public Sewer
          </td>
          <td
            key="1"
          >
            3143777
          </td>
          <td
            key="2"
          >
            Moore
          </td>
        </tr>
        <tr
          key="2"
        >
          <th
            key="2"
            scope="row"
          >
            3
          </th>
          <td
            key="0"
          >
            Public Sewer
          </td>
          <td
            key="1"
          >
            3228726
          </td>
          <td
            key="2"
          >
            Wagner
          </td>
        </tr>
        <tr
          key="3"
        >
          <th
            key="3"
            scope="row"
          >
            4
          </th>
          <td
            key="0"
          >
            Public Sewer
          </td>
          <td
            key="1"
          >
            3158743
          </td>
          <td
            key="2"
          >
            Masih
          </td>
        </tr>
        <tr
          key="4"
        >
          <th
            key="4"
            scope="row"
          >
            5
          </th>
          <td
            key="0"
          >
            Public Sewer
          </td>
          <td
            key="1"
          >
            78774
          </td>
          <td
            key="2"
          >
            Gatewood
          </td>
        </tr>
      </tbody>
    </Table>
  </ModalBody>
  <ModalFooter
    tag="div"
  >
    <Button
      color="primary"
      onClick={[Function]}
      tag="button"
    >
      Close
    </Button>
  </ModalFooter>
</Modal>
`;
